src/almanack/batch.py

```diff
@@ -0,0 +1,235 @@
+"""
+Batch processing utilities for running Almanack across many repositories.
+"""
+
+import json
+import logging
+from concurrent.futures import Executor, ProcessPoolExecutor, as_completed
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Sequence, Type, Union
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+from almanack.metrics.data import process_repo_for_almanack
+
+RepositoryProcessor = Callable[[str], dict[str, Any]]
+LOGGER = logging.getLogger(__name__)
+
+
+def sanitize_for_parquet(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Cleans a DataFrame so all columns are parquet-safe.
+
+    - Expands dict columns into multiple fields.
+    - Converts lists into JSON strings.
+    - Casts generic object types into strings.
```

Comment:
how will these bullets render in an autodoc? I'm not sure if this will look as cleanly as simply expanding this into sentences.